filters/crop/Point.cpp

```diff
@@ -0,0 +1,189 @@
+/******************************************************************************
+* Copyright (c) 2016, Howard Butler (howard@hobu.co)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "Point.hpp"
+
+namespace pdal
+{
+
+namespace
+{
+
+const double LOWEST = (std::numeric_limits<double>::lowest)();
+const double HIGHEST = (std::numeric_limits<double>::max)();
+
+}
+
+namespace cropfilter
+{
+
+Point::Point()
+    : Geometry()
+    , x(LOWEST)
+    , y(LOWEST)
+    , z(LOWEST)
+{
+
+};
+
+Point::Point(const std::string& wkt_or_json, SpatialReference ref)
+    : Geometry(wkt_or_json, ref)
+{
+
+}
+
+void Point::update(const std::string& wkt_or_json, SpatialReference ref)
+{
+    bool isJson = wkt_or_json.find("{") != wkt_or_json.npos ||
```

Comment:
`npos` is a [static value](http://en.cppreference.com/w/cpp/string/basic_string/npos), a bit strange to see it called from an instance.